sky/clouds/kubernetes.py

```diff
@@ -164,15 +164,20 @@ def existing_allowed_contexts(cls, silent: bool = False) -> List[str]:
 
         all_contexts = set(all_contexts)
 
+        # Allowed_contexts specified for workspace should take precedence over
+        # the global allowed_contexts.
+        allowed_contexts = skypilot_config.get_workspace_cloud(
+            'kubernetes').get('allowed_contexts', None)
```

Comment:
I'm a bit uneasy about devs having to query two sources of config. `skypilot_config.get_nested` should account for the current active workspace (i.e., workspace becomes an override in the hierarchy).